blockmanager.go

```diff
@@ -595,6 +595,96 @@ func (b *blockManager) handleDonePeerMsg(peers *list.List, sp *serverPeer) {
 	}
 }
 
+// errToWireRejectCode determines the wire rejection code and description for a
+// given error. This function can convert some select blockchain and mempool
+// error types to the historical rejection codes used on the p2p wire protocol.
+func errToWireRejectCode(err error) (wire.RejectCode, string) {
+	if err == nil {
+		panic("errToWireRejectCode(): err should not be nil")
```

Comment:
This is a remote exploit waiting to happen.

I describe in a bit more detail below, but in general, panics in error paths, especially those that are dealing with external data are incredibly dangerous.  This case isn't quite as bad as the one I discuss below, but this also needs to be changed to return a generic error instead.